src/metabase/lib/join.cljc

```diff
@@ -491,17 +507,20 @@
                          (= fk-target-field-id pk-id)))
                   (lib.metadata.calculation/visible-columns query stage-number (lib.util/query-stage query stage-number)))))
 
-(mu/defn suggested-join-condition :- [:maybe ::lib.schema.external-op/external-op]
-  "Return a suggested default join condition when constructing a join against `joined-thing`, e.g. a Table, Saved
+(mu/defn suggested-join-condition :- [:maybe ::lib.schema.expression/boolean] ; i.e., a filter clause
+  "Return a suggested default join condition when constructing a join against `joinable`, e.g. a Table, Saved
   Question, or another query. A suggested condition will be returned if the query stage has a foreign key to the
   primary key of the thing we're joining (see #31175 for more info); otherwise this will return `nil` if no default
   condition is suggested."
-  ([query joined-thing]
-   (suggested-join-condition query -1 joined-thing))
+  ([query joinable]
+   (suggested-join-condition query -1 joinable))
 
   ([query         :- ::lib.schema/query
     stage-number  :- :int
-    joined-thing]
-   (when-let [pk-col (pk-column query stage-number joined-thing)]
-     (when-let [fk-col (fk-column query stage-number pk-col)]
-       (lib.filter/filter-clause (equals-join-condition-operator-definition) fk-col pk-col)))))
+    joinable]
+   (when-let [pk-col (pk-column query stage-number joinable)]
+     (when-let [fk-col (fk-column-for query stage-number pk-col)]
+       (lib.common/->op-arg
+        query
+        stage-number
+        (lib.filter/filter-clause (equals-join-condition-operator-definition) fk-col pk-col))))))
```

Comment:
Once https://github.com/metabase/metabase/issues/31676 is fixed, we won't need to use `->op-arg` to convert the external op back to a raw clause.